zebra-network/src/protocol/external/message.rs

```diff
@@ -340,25 +340,47 @@ pub enum Message {
     FilterClear,
 }
 
+/// The maximum size of the rejection message.
+///
+/// This is equivalent to `COMMAND_SIZE` in zcashd.
+const MAX_REJECT_MESSAGE_LENGTH: usize = 12;
+
+/// The maximum size of the rejection reason.
+///
+/// This is equivalent to `MAX_REJECT_MESSAGE_LENGTH` in zcashd.
+const MAX_REJECT_REASON_LENGTH: usize = 111;
+
+// TODO: add tests for Error conversion and Reject message serialization (#4633)
+// (Zebra does not currently send reject messages, and it ignores received reject messages.)
 impl<E> From<E> for Message
 where
     E: Error,
 {
     fn from(e: E) -> Self {
+        let message = e
+            .to_string()
+            .escape_default()
+            .take(MAX_REJECT_MESSAGE_LENGTH)
+            .collect();
```

Comment:
I didn't realise that `escape` returned an iterator.

This allocates the same amount of memory for the temporary string, but it's dropped right after the truncated string is created.

Reply:
lookg good. thanks.